plugin/pkg/managedseed/validator/admission.go

```diff
@@ -0,0 +1,463 @@
+	}
+	if ms != nil && ms.Name != managedSeed.Name {
+		return apierrors.NewInvalid(gk, managedSeed.Name, append(allErrs, field.Invalid(shootNamePath, managedSeed.Spec.Shoot.Name, fmt.Sprintf("shoot %s already registered as seed", kutil.ObjectName(shoot)))))
+	}
+
+	switch {
+	case managedSeed.Spec.SeedTemplate != nil:
+		// Admit seed spec against shoot
+		errs, err := v.admitSeedSpec(ctx, &managedSeed.Spec.SeedTemplate.Spec, shoot, field.NewPath("spec", "seedTemplate", "spec"))
+		if err != nil {
+			return err
+		}
+		allErrs = append(allErrs, errs...)
+
+	case managedSeed.Spec.Gardenlet != nil:
+		// Admit gardelnet against shoot
+		errs, err := v.admitGardenlet(ctx, managedSeed.Spec.Gardenlet, shoot, field.NewPath("spec", "gardenlet"))
+		if err != nil {
+			return err
+		}
+		allErrs = append(allErrs, errs...)
+	}
+
+	if len(allErrs) > 0 {
+		return apierrors.NewInvalid(gk, managedSeed.Name, allErrs)
+	}
+
+	return nil
+}
+
+func (v *ValidateManagedSeed) admitGardenlet(ctx context.Context, gardenlet *seedmanagement.Gardenlet, shoot *gardencore.Shoot, fldPath *field.Path) (field.ErrorList, error) {
+	var allErrs field.ErrorList
+
+	if gardenlet.Config != nil {
+		configPath := fldPath.Child("config")
+
+		// Decode gardenlet config to an internal version
+		// Without defaults, since we don't want to set gardenlet config defaults in the resource at this point
+		gardenletConfig, err := helper.ConvertGardenletConfigExternal(gardenlet.Config)
+		if err != nil {
+			return append(allErrs, field.Invalid(configPath, gardenlet.Config, fmt.Sprintf("could not decode config: %v", err))), nil
+		}
+
+		if gardenletConfig.SeedConfig != nil {
+			seedConfigPath := fldPath.Child("seedConfig")
+
+			// Convert seed config to internal version
+			seed, err := helper.ConvertSeed(&gardenletConfig.SeedConfig.Seed)
+			if err != nil {
+				return allErrs, apierrors.NewInternalError(fmt.Errorf("could not convert seed config: %v", err))
+			}
+
+			// Admit seed spec against shoot
+			errs, err := v.admitSeedSpec(ctx, &seed.Spec, shoot, seedConfigPath.Child("spec"))
+			if err != nil {
+				return allErrs, err
+			}
+			allErrs = append(allErrs, errs...)
+
+			// Set seed config back to gardenletConfig.SeedConfig.Seed
+			seedx, err := helper.ConvertSeedExternal(seed)
+			if err != nil {
+				return allErrs, apierrors.NewInternalError(fmt.Errorf("could not convert seed config: %v", err))
+			}
+			gardenletConfig.SeedConfig.Seed = *seedx
+		}
+
+		// Set gardenlet config back to gardenlet.Config
+		gardenlet.Config = gardenletConfig
+	}
+
+	return allErrs, nil
+}
+
+func (v *ValidateManagedSeed) admitSeedSpec(ctx context.Context, spec *gardencore.SeedSpec, shoot *gardencore.Shoot, fldPath *field.Path) (field.ErrorList, error) {
+	var allErrs field.ErrorList
+
+	// Initialize backup provider
+	if spec.Backup != nil && spec.Backup.Provider == "" {
```

Comment:
What you described is mostly how it works today: `backup: {}` results in a fully and correctly initialized `backup` section in `seedConfig`, in `seedTemplate` you would also need to specify an empty `secretRef`, like this:
```yaml
backup:
  secretRef: {}
```
The difference comes from the fact that `gardenlet.config` is a raw extension and therefore not validated by the API server before the admission plugin kicks in, while `seedTemplate` is strongly typed and is validated.

In both cases, don't think `omitempty` plays any role.

Reply:
Yes, for the `seedConfig` case it works because of the raw extension, right.
If we would remove the `omitempty` tag for the `backup` field, one could also set `backup: {}` just like in the `seedConfig` case. 
I think, this would make it consistent and avoid confusion. WDYT?

Reply:
I removed `omitempty`, it doesn't work as you suggest it would. With a resource that has `backup: {}` in `seedTemplate` I still get the same error as before:
```
$ k apply -f dev/managedseed-i500152-gcpx-ng.yaml
error: error validating "dev/managedseed-i500152-gcpx-ng.yaml": error validating data: [ValidationError(ManagedSeed.spec.seedTemplate.spec.backup): missing required field "provider" in com.github.gardener.gardener.pkg.apis.core.v1beta1.SeedBackup, ValidationError(ManagedSeed.spec.seedTemplate.spec.backup): missing required field "secretRef" in com.github.gardener.gardener.pkg.apis.core.v1beta1.SeedBackup]; if you choose to ignore these errors, turn validation off with --validate=false
```

In fact, I was also wrong, what you need for an empty backup in `seedTemplate` is:
```yaml
backup:
  provider: ""
  secretRef: {}
```

So you must specify these empty values, but they are overwritten by the admission / defaults mechanism to appropriate ones.

Why do you think `omitempty` should play any role? It only controls how values are serialized to JSON. How could it influence such things as you suggest?

As far as I can tell, for `backup: {}` to work we would need to make both `provider` and `secretRef` inside `backup` non-mandatory. Not sure if this is possible, but even if it is, the value of being able to specify just `backup: {}` instead of the above seems too small to warrant an incompatible API change.

Reply:
Ok, let's keep it like this for now, as it seems to be a bit more complicated.
The main usecase with the gardenletConfig works...